### PR TITLE
106 - migrate mainnet stats to monorepo

### LINKED ISSUE #106: Migrate mainnet-stats into monorepo

### PR DESCRIPTION
**Pull request by sam-keen**
Closes #106

### CODE REVIEW

File: apps/stats-mainnet/src/styles/styles.css [hunk 1]
```diff
@@ -0,0 +1,31 @@
+@tailwind base;
+@tailwind components;
+
+@font-face {
+  font-family: 'AlphaLyrae-Medium';
+  src: url('./AlphaLyrae-Medium.woff');
```
Comment (dexturr) [inline apps/stats-mainnet/src/styles/styles.css:6]: follow up (non-blocking): we need to add alpha lyrae support to the UI toolkit

File: apps/stats-mainnet/src/index.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      // Set dark mode preference indicator here to avoid FOUC
+      if (
+        localStorage.theme === 'dark' ||
+        (!('theme' in localStorage) &&
+          window.matchMedia('(prefers-color-scheme: dark)').matches)
+      ) {
+        document.documentElement.classList.add('dark-mode-preferred');
+      } else {
+        document.documentElement.classList.remove('dark-mode-preferred');
+      }
```
Comment (dexturr) [inline apps/stats-mainnet/src/index.html:35]: chore (blocking): A hook now exists for this

File: apps/stats-mainnet/src/components/images/light-mode-toggle.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export const LightModeToggle = () => {
```
Comment (dexturr) [inline apps/stats-mainnet/src/components/images/light-mode-toggle.tsx:1]: chore (blocking): Use UI toolkit switcher that now exists

File: apps/stats-mainnet-e2e/cypress.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (dexturr) [inline apps/stats-mainnet-e2e/cypress.json:1]: chore (blocking): make cypress tests run on a port that is not going to be used by other tests. Currently in use:
* 3000
* 4200

Comment (dexturr) [reply apps/stats-mainnet-e2e/cypress.json:1]: You can compare this to either BE or tradiung, one of them is using non-default ports for the config

File: apps/stats-mainnet-e2e/cypress.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  "fileServerFolder": ".",
```
Comment (dexturr) [inline apps/stats-mainnet-e2e/cypress.json:2]: follow up (non-blocking): let's create a ticket to add some cypress tests for this

File: apps/stats-mainnet/src/styles/styles.css (hunk 1 above)
Comment (sam-keen) [reply apps/stats-mainnet/src/styles/styles.css:6]: Yes indeedy. Bart and I were talking about that this morning, we're going to find a way to have it accessible for the whole monorepo

File: apps/stats-mainnet/.browserslistrc [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# This file is used by:
+# 1. autoprefixer to adjust CSS to support the below specified browsers
+# 2. babel preset-env to adjust included polyfills
+#
+# For additional information regarding the format and rule options, please see:
+# https://github.com/browserslist/browserslist#queries
+#
+# If you need to support different browsers in production, you may tweak the list below.
+
+last 1 Chrome version
```
Comment (dexturr) [inline apps/stats-mainnet/.browserslistrc:10]: follow up (non-blocking): need a meeting to decide our browser support. THis does not include brave, which I would consider probably in our list of supported browsers. Probably will still work as it's backed onto Chrome V8 but something we should nail down as a team

File: apps/stats-mainnet/src/index.html (hunk 2 above)
Comment (sam-keen) [reply apps/stats-mainnet/src/index.html:35]: Does the hook avoid a FOUC? I did this intentionally at this early stage

File: apps/stats-mainnet/src/config/types.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  blockDuration: string;
+  status: string;
+  vegaTime: string;
+  appVersion: string;
+  chainVersion: string;
+  uptime: string;
+  chainId: string;
+}
+
+export type value = any;
```
Comment (dexturr) [inline apps/stats-mainnet/src/config/types.ts:21]: question (non-blocking): can we do:
```suggestion
export type value = string | number | null | undefined;
```

File: apps/stats-mainnet-e2e/cypress.json (hunk 5 above)
Comment (dexturr) [reply apps/stats-mainnet-e2e/cypress.json:2]: #109

File: apps/stats-mainnet/src/index.html (hunk 2 above)
Comment (dexturr) [reply apps/stats-mainnet/src/index.html:35]: It does not avoid a FOUC, but it should be up to the hook to fix it. 

Happy to ticket this as a follow up in that case if fixing the hook would be a lot of work and we don't want a FOUC right now. I think @campbellssource noticed this when we were looking at the Trading UI on Friday?

File: apps/stats-mainnet/src/styles/styles.css (hunk 1 above)
Comment (dexturr) [reply apps/stats-mainnet/src/styles/styles.css:6]: #110

File: apps/stats-mainnet/src/index.html (hunk 2 above)
Comment (sam-keen) [reply apps/stats-mainnet/src/index.html:35]: Sounds good, I'd like to get this on block explorer asap

File: apps/stats-mainnet/src/components/images/light-mode-toggle.tsx (hunk 3 above)
Comment (sam-keen) [reply apps/stats-mainnet/src/components/images/light-mode-toggle.tsx:1]: This isn't currently compatible with the way I've set up the dark mode, will deal with this as part of #113

File: apps/stats-mainnet/.browserslistrc (hunk 6 above)
Comment (sam-keen) [reply apps/stats-mainnet/.browserslistrc:10]: Good shout

File: libs/ui-toolkit/src/components/vega-logo/vega-logo.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { render } from '@testing-library/react';
```
Comment (dexturr) [inline libs/ui-toolkit/src/components/vega-logo/vega-logo.spec.tsx:1]: Use screen ;) MWAHAHAHA

**Review by dexturr — APPROVED** [review]

File: libs/mainnet-stats-manager/src/components/good-threshold-indicator/good-threshold-indicator.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+export const GoodThresholdIndicator = ({
+  goodThreshold,
+  value,
+}: GoodThresholdIndicatorProps) => {
+  return (
+    <div
+      className={`inline-block w-8 h-8 mb-[0.15rem] mr-8 rounded ${
+        (goodThreshold &&
+          (goodThreshold(value) ? 'bg-intent-success' : 'bg-intent-danger')) ||
```
Comment (mattrussell36) [inline libs/mainnet-stats-manager/src/components/good-threshold-indicator/good-threshold-indicator.tsx:16]: I don't see these classes used or imported anywhere else?

**Review by mattrussell36 — APPROVED** [review]
LGTM. I'd prefer to not tie the stats app and lib to mainnet by the name stats-mainnet and mainnet-stats-manager. Can this be changed so that we can specify the environment using .env. Happy for that to be a follow up though.

File: libs/mainnet-stats-manager/src/components/good-threshold-indicator/good-threshold-indicator.tsx (hunk 9 above)
Comment (sam-keen) [reply libs/mainnet-stats-manager/src/components/good-threshold-indicator/good-threshold-indicator.tsx:16]: They're in the theme, I think this is the intended use for them

**Pull request merged**